data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3SourceConfig.java

```diff
@@ -28,7 +28,7 @@ public class S3SourceConfig {
     @NotNull
     private NotificationTypeOption notificationType;
 
-    @JsonProperty("compression")
+    @JsonProperty("compression_type")
```

Comment:
This is a breaking change isn't it? We need to allow both configuration options and deprecate "compression"

Reply:
Thanks for catching this @graytaylor0 .

I don't think we want to alter this `compression` option at all.

If anything, we could use this value in the `s3_select` configuration instead of the new `compression_type`. This could look like the following:

```
s3:
  compression: gzip
  s3_select:
    expression: "SELECT *"
```

But, I'm not sure this is really going to simplify the user experience or not. The set of allowable values may change between the two and that could lead to confusion.

Reply:
Thanks Taylor for your review, we are reverted this value to **compression** as per @dlvenable suggestion.